pprint/shared/src/main/scala/pprint/PPrint.scala

```diff
@@ -447,9 +447,13 @@ object Internals {
               q"""
               $pkg.PPrint[$targetType](
                 $pkg.PPrinter.ChunkedRepr[$targetType](
-                  $pkg.Chunker[$targetType]( ($t: $targetType, $cfg: $pkg.Config) =>
-                    $pkg.Chunker.$tupleChunker[..$argTypes].chunk($w, $cfg)
-                  )
+                  $pkg.Chunker[$targetType]{ ($t: $targetType, $cfg: $pkg.Config) =>
+                    val chunkedArgs = $pkg.Chunker.$tupleChunker[..$argTypes].chunk($w, $cfg)
+                    if ($cfg.showFieldNames) chunkedArgs
+                      .zip(Iterator.apply(..${argSyms.map(_.name.toString + " = ")}))
+                      .map{case (chunkedArg, field) => Iterator.single(field) ++ chunkedArg}
+                    else chunkedArgs
```

Comment:
Can you extract this into a function? In general the code we generate inside macros should be minimal and any non-trivial logic like this stuff should end up in a helper function somewhere.